src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -289,33 +292,99 @@ void SignalDetectorClass::processMessage()
 				//postamble = "";
 
 				/*				Output raw message Data				*/
-				MSG_PRINT(MSG_START); 				
-				MSG_PRINT("MS"); MSG_PRINT(SERIAL_DELIMITER);
-				for (uint8_t idx = 0; idx < patternLen; idx++)
-				{
-					if (pattern[idx] == 0 || histo[idx] == 0) continue;
-					MSG_PRINT('P'); MSG_PRINT(idx); MSG_PRINT('='); MSG_PRINT(pattern[idx]); MSG_PRINT(SERIAL_DELIMITER);
-				}
-				MSG_PRINT('D=');
+				if (MredEnabled) {
+					int patternInt;
+					uint8_t patternLow;
+					uint8_t patternIdx;
+					
+					MSG_PRINT(MSG_START);  MSG_PRINT("Ms");  MSG_PRINT(SERIAL_DELIMITER);
+					for (uint8_t idx = 0; idx < patternLen; idx++)
+					{
+						if (pattern[idx] == 0 || histo[idx] == 0) continue;
+						patternIdx = idx;
+						patternInt = pattern[idx];
 
-				for (uint8_t i = mstart; i <= mend; ++i)
-				{
-					MSG_PRINT(message[i]);
-				}
-				MSG_PRINT(SERIAL_DELIMITER);
-				MSG_PRINT("CP="); MSG_PRINT(clock);     MSG_PRINT(SERIAL_DELIMITER);     // ClockPulse
-				MSG_PRINT("SP="); MSG_PRINT(sync);      MSG_PRINT(SERIAL_DELIMITER);     // SyncPulse
-				MSG_PRINT("R=");  MSG_PRINT(rssiValue); MSG_PRINT(SERIAL_DELIMITER);     // Signal Level (RSSI)
+						if (patternInt < 0) {
+							patternIdx = idx | B10100000;    // Bit5 = 1 (Vorzeichen negativ)
+							patternInt = -patternInt;
+						}
+						else {
+							patternIdx = idx | B10000000;    // Bit5 = 0 (Vorzeichen positiv)
+						}
+						
+						patternLow = lowByte(patternInt);
+						if (bitRead(patternLow,7) == 0) {
```

Comment:
dieses Macro kannte ich noch nicht 👍 , aber wir könnte das 7, bit doch einfach immer auf 1 setzten, auch wenn es bereits auf 1 gesetzt ist. Das würde die Abfrage überflüssig machen, oder übersehe ich etwas?
Bit 4 von idx soll wohl auch immer auf den gleichen Wert wie bit7 von patternLow stehen? Wobei ich angenommen hätte, das Vorzeichen steckt in Bit 15 von patternInt also patternInt >>15 entspricht dem Vorzeichen.

`patternIdx &= (patternInt >>15) <<4    ` würde dann bit 4 auf den gleich Wert wie Bit 15 aus patternInt setzen.